pkg/chunk/elastic/es_table_client.go

```diff
@@ -0,0 +1,99 @@
+package elastic
+
+import (
+	"context"
+	"github.com/olivere/elastic"
+
+	"github.com/pkg/errors"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+)
+
+//var ctx = context.Background()
+
+type tableClient struct {
+	cfg    Config
+	client *elastic.Client
+}
+
+// NewTableClient returns a new TableClient.
+func NewTableClient(ctx context.Context, cfg Config) (chunk.TableClient, error) {
+	client, err := newES(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &tableClient{
+		cfg:    cfg,
+		client: client,
+	}, nil
+}
+
+// ListTables means list index in ElasticSearch
+func (c *tableClient) ListTables(ctx context.Context) ([]string, error) {
+	response, err := c.client.IndexNames()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return response, nil
+}
+
+func (c *tableClient) CreateTable(ctx context.Context, desc chunk.TableDesc) error {
+	// Here we use ElasticSearch auto index management with templates, no need to create table manually.
+	// Instead, we create Template.
+	templateName := "loki-index"
+	// check if template exists, and create one if not exist
+	exists, err := c.client.IndexTemplateExists(templateName).Do(ctx)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	if !exists {
+		tmpl := `{
+		"index_patterns":["lokiindex_*"],
```

Comment:
this template pattern should not be hard coded. Also this seems to be leaking a loki detail into cortex.